src/app/dwg/qgsdwgimportdialog.cpp

```diff
@@ -124,11 +123,31 @@ void QgsDwgImportDialog::updateUI()
 
 void QgsDwgImportDialog::pbBrowseDatabase_clicked()
 {
-  QString dir( leDatabase->text().isEmpty() ? QDir::homePath() : QFileInfo( leDatabase->text() ).canonicalPath() );
+  QgsSettings s;
+  QString dir( s.value( QStringLiteral( "/DwgImport/lastDirDatabase" ), QDir::homePath() ).toString() );
   QString filename = QFileDialog::getSaveFileName( this, tr( "Specify GeoPackage database" ), dir, tr( "GeoPackage database" ) + " (*.gpkg *.GPKG)", nullptr, QFileDialog::DontConfirmOverwrite );
   if ( filename.isEmpty() )
     return;
-  leDatabase->setText( filename );
+
+  QFileInfo fi( filename );
+  if ( fi.exists() && ( QMessageBox::question( this,
+                        tr( "File exists" ),
+                        tr( "The file already exists. Do you want to overwrite the existing file?" ),
+                        QMessageBox::Ok | QMessageBox::Cancel ) == QMessageBox::Cancel ) )
+  {
+    leDatabase->clear();
+    return;
+  }
+  else
+  {
+
+    if ( !( filename.endsWith( ".gpkg" ) || filename.endsWith( ".GPKG" ) ) )
```

Comment:
Use QgsFileUtils::ensureFileNameHasExtension() instead.

Reply:
Ultimately it'd be nice to use QgsFileWidget here, but this that's understandably more work...